### PR TITLE
Add check for cluster resources failures in testsetup

### PR DESCRIPTION
**Pull request by vmoravec**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -3698,6 +3698,9 @@ function oncontroller_testsetup
 {
     . .openrc
     oncontroller_prepare_functional_tests
+    if [[ $hacloud = 1 ]] ; then
+        crm_mon --failcounts -1 | grep "fail-count=" && complain 34 "Cluster resources' failures detected"
```
Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: I'm not sure that's right.  At least it doesn't catch failures like this:

```
Failed Actions:
* nova-compute_monitor_10000 on remote-d52-54-77-77-77-03 'not running' (7): call=13560, status=complete, exitreason='none',
    last-rc-change='Thu Feb  2 17:08:13 2017', queued=0ms, exec=0ms
```

Please can you update the commit message to explain which types of errors it *does* catch, so we can compare the two and decide whether it's also important to catch this one?

Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: The goal is to start low. Having a few errors that are definitely errors caught is much better than having nothing checked and fail counts exceeded already after the initial deployment. 

If you look at any ha log for cloud7, you see failures due to chef trying to instantiate resources that are not yet installed from packages at that point. Which is a trivial error and that should not creep in anymore.

Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: LGTM but 34 is also used to signalize missing media

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Someone should really go though all those mkcloud exit codes and make sure they are unique. I believe there are much more cases than this :-)

Comment (vmoravec) [inline scripts/qa_crowbarsetup.sh]: @aplanas Will change it to 55, thanks for spotting this.

Comment (vmoravec) [inline scripts/qa_crowbarsetup.sh]: @aspiers TBH I don't know how broad the error coverage of `crm_mon` is, I was expecting to catch all  cases. As Dirk said, it's the first step, we can tune it later.

Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: @vmoravec Do you mean you *weren't* expecting to catch all cases?

@dirkmueller @vmoravec I'm fine with starting low as long as we don't forget to raise the bar higher later on :)

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: @aplanas, @jsuchome, @vmoravec 
The exit codes are not meant to be unique (we don't have enough to make them unique), just seldom. If you print a message with complain anyway, you can quickly find the error.
If its done via safely you need to check the command on the previous line(s).

**Pull request merged**